### PR TITLE
Fixed #7621 - Add support for config_override.test.php

### LINKED ISSUE #7621: Add support for config_override.test.php
#### Issue
This is part of #7344, but I'm splitting it out into a separate issue.

Add support for a `config_override.test.php` file, which would be used to define configuration values for the test environment specifically (e.g. so you can run the tests on a separate database from the development site).

There are certain config options you'd want to use specifically for the test environment, like `imap_test` or statechecker-related options. It'd be nice to have a `config_override.test.php` which would only be used when the tests are being run, to allow separating the development and test environments.

I'm not sure if the `config_override.php` should still be loaded in the test environment, I guess it should be? It'd load `config.php`, then `config_override.php` which would override any settings in `config.php`, then `config_override.test.php` which would override any settings in `config_override.php`. When not running the tests, the `config_override.test.php` would be ignored.

[Rails comes with separate environments for development, test, and production](https://guides.rubyonrails.org/configuring.html#creating-rails-environments), for example. Laravel works the same way. I think it's a pretty smart pattern, and makes it easier to write code and run tests at the same time, without one messing with the other.

### PR DESCRIPTION
**Pull request by Dillon-Brown**
## Motivation and Context

Issue reference: #7621 

## How To Test This

1. Check Travis
2. Add a ```tests/config.test.php``` file and then give it a custom value like so: 
```
$testConfig = [
    'imap_test' => true,
];
```
3. Check that the custom config only runs during the test suite execution and overrides the ```config.php``` and ```config_override.php``` values.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist

- [X] My code follows the code style of this project found [here](https://docs.suitecrm.com/community/contributing-code/coding-standards/).
- [X] My change requires a change to the documentation.
- [X] I have read the [**How to Contribute**](https://docs.suitecrm.com/community/contributing-code/) guidelines.

### CODE REVIEW

**Comment by connorshea** [comment]
We should probably have a `config.test.example.php` or `config.test.dist.php` file in the repo for users to copy from. :)

Just this should be fine? Maybe a comment explaining what it's for.

```php
<?php
$testConfig = [
    'imap_test' => true,
];
```

This'll be great, thank you! ❤️

**Comment by cameronblaikie** [comment]
Assessed :+1:

**Comment by cameronblaikie** [comment]
Needs input from @Dillon-Brown before merging :D.

**Comment by cameronblaikie** [comment]
Assessed :+1:

File: lib/Robo/Traits/CliRunnerTrait.php [hunk 1]
```diff
@@ -65,13 +65,27 @@ protected function bootstrap()
...
         $root = __DIR__ . '/../../../';
 
         require $root . 'config.php';
         require $root . 'config_override.php';
         require_once $root . 'include/entryPoint.php';
 
+        // Load up the config.test.php file. This is used to define configuration values for the test environment.
+        $testConfig = [];
+
+        if (is_file($root . 'tests/config.test.php')) {
```
Comment (connorshea) [inline lib/Robo/Traits/CliRunnerTrait.php]: it occurs to me now, wouldn't this effect all the Robo tasks that use CliRunnerTrait?

Comment (connorshea) [reply lib/Robo/Traits/CliRunnerTrait.php]: Also: when I run `./vendor/bin/robo tests:unit`, at least on the StateChecker-removal branch, the `config.php` file is overwritten by the contents of the `config.test.php`.

**Pull request merged**